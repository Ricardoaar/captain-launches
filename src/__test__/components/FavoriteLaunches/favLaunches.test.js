import FavoriteLaunches from "../../../components/FavoritesLaunches";
import { shallow } from "enzyme";
import { create } from "react-test-renderer";
import React from "react";

const launches = require("../../__mocks__/launches.mock.js");

const MockRender = (launch) => (
  <div key={launch.mission}>
    {launch.mission}
    {launch.date}
  </div>
);

describe("<FavoritesLaunches/>", () => {
  const wrapper = shallow(
    <FavoriteLaunches render={MockRender} favorites={[launches[0], launches[1]]} />
  );
  test("Should render successfully", () => {
    expect(wrapper.length).toBe(1);
  });

  test("Should render just one wrapper", () => {
    expect(wrapper.find(".favorites-wrapper").length).toBe(1);
  });
  test("should render just one favorite container", () => {
    expect(wrapper.find(".favorites-container").length).toBe(1);
  });
  test("should match the snapshot", () => {
    const wrapper = create(
      <FavoriteLaunches render={MockRender} favorites={[launches[0], launches[1]]} />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
