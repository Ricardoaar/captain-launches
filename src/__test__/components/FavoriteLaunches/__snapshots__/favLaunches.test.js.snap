// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavoritesLaunches/> should match the snapshot 1`] = `
<section
  className="row-span-5 flex flex-col mt-20 space-top-20 h-full"
>
  <h3
    className="page-title"
  >
    Favorites Launches
  </h3>
  <div
    className="favorites-wrapper overflow-y-hidden"
  >
    <div
      className="favorites-container"
    >
      <div>
        Save the world
      </div>
      <div>
        Save the other world
      </div>
    </div>
  </div>
</section>
`;
